src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -1823,6 +2047,11 @@ private static ExtendedErrorTypeSymbol CyclicInheritanceError(TypeSymbol declare
 
         private NamedTypeSymbol MakeAcyclicBaseType()
         {
+            if (TypeKind is TypeKind.Extension)
```

Comment:
>if (TypeKind is TypeKind.Extension) [](http://example.com/codeflow?start=12&length=35)

Should this affect ```GetDeclaredBaseType``` too?

Reply:
If we put similar logic in `GetDeclaredBaseType`, then we create a cycle in `TypeKind`.
My thinking is that "declared" APIs don't lie, they show the reality of the underlying metadata. We can make adjustments on top of that where needed.

Reply:
> My thinking is that "declared" APIs don't lie, they show the reality of the underlying metadata.

Do we return System.ValueType as declared base for source symbol representing extension type?

Reply:
Ah, good point. We return `null` for source symbols. Thanks